### PR TITLE
Fix footer style according to Design Library

### LINKED ISSUE #484: Some details on footer don't follow the Figma designs
## Description
Inconsistencies are:
- Menu items don't have "Heading 4B" style
- Text of newsletter input (value & placeholder) is hardly visible
- Text and icon of donate button has an underline on hover

## Reproduction

1. See the [Footer story](https://cc-vocabulary.netlify.app/?path=/story/patterns-footer--default-story).

## Expectation
Follow styles of the [Design Library](https://www.figma.com/file/l4Mt3dn3Ndtrvrb4aLcwXI/Design-Library?node-id=639%3A185).

## Resolution

- [x] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #484 by @krysal 

## Description
Change styles mentioned in the issue.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/stories/footer.stories.mdx [hunk 1]
```diff
@@ -54,7 +54,8 @@ export const footer = () => `
             <div class="subscription">
               <h5 class="b-header">Subscribe to our newsletter</h5>
               <form class="newsletter">
-                <input type="text" class="input" placeholder="Your email">
+                <label for="email" class="is-sr-only"></label>
```
Comment (dhruvkb) [inline src/stories/footer.stories.mdx]: If you have a label for screen-readers, please also add the text that the screen-reader should read.
```suggestion
                <label for="email" class="is-sr-only">Email to subscribe</label>
```

File: src/styles/footer.scss [hunk 2]
```diff
@@ -63,6 +62,11 @@
       .input {
         background-color: transparent;
         border-color: $color-dark-gray;
+        color: $color-dark-gray;
+          &::placeholder {
+            color: $color-dark-gray;
+            opacity: 0.8;
+          }
```
Comment (dhruvkb) [inline src/styles/footer.scss]: Indentation is a bit off.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

**Comment by dhruvkb** [comment]
Sorry for the late review @krysal.

File: src/stories/footer.stories.mdx (hunk 1 above)
Comment (krysal) [reply src/stories/footer.stories.mdx]: Ups, of course, I forgot it, my bad.

File: src/styles/footer.scss (hunk 2 above)
Comment (krysal) [reply src/styles/footer.scss]: I see no problem here, it's two spaces like the rest of the file, how should it be?

Comment (krysal) [reply src/styles/footer.scss]: Oh, I already noticed the whole block has an extra indentation, fixing.

**Comment by krysal** [comment]
It's okay! I applied the required corrections.

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**